tests/describe_task_queue_test.go

```diff
@@ -0,0 +1,230 @@
+		dynamicconfig.FrontendNamespaceReplicationInducingAPIsRPS:                               1000,
+
+		// The dispatch tests below rely on being able to see the effects of changing
+		// versioning data relatively quickly. In general, we only promise to act on new
+		// versioning data "soon", i.e. after a long poll interval. We can reduce the long poll
+		// interval so that we don't have to wait so long.
+		// TODO: update this comment. it may be outdated and/or misleading.
+		dynamicconfig.MatchingLongPollExpirationInterval: longPollTime,
+
+		// this is overridden for tests using testWithMatchingBehavior
+		dynamicconfig.MatchingNumTaskqueueReadPartitions:  4,
+		dynamicconfig.MatchingNumTaskqueueWritePartitions: 4,
+	}
+	s.setupSuite("testdata/es_cluster.yaml")
+}
+
+func (s *DescribeTaskQueueSuite) TearDownSuite() {
+	s.tearDownSuite()
+}
+
+func (s *DescribeTaskQueueSuite) SetupTest() {
+	// Have to define our overridden assertions in the test setup. If we did it earlier, s.T() will return nil
+	s.Assertions = require.New(s.T())
+	s.ProtoAssertions = protorequire.New(s.T())
+	s.HistoryRequire = historyrequire.New(s.T())
+	s.UpdateUtils = updateutils.New(s.T())
+}
+
+func (s *DescribeTaskQueueSuite) TestAddNoTasks_ValidateBacklogInfo() {
+	s.PublishConsumeWorkflowTasksValidateBacklogInfo(4, 0)
+}
+
+func (s *DescribeTaskQueueSuite) TestAddSingleTask_ValidateBacklogInfo() {
+	s.PublishConsumeWorkflowTasksValidateBacklogInfo(1, 1)
+}
+
+func (s *DescribeTaskQueueSuite) TestAddMultipleTasksMultiplePartitions_ValidateBacklogInfo() {
+	s.PublishConsumeWorkflowTasksValidateBacklogInfo(4, 4)
+}
+
+func (s *DescribeTaskQueueSuite) PublishConsumeWorkflowTasksValidateBacklogInfo(partitions int, workflows int) {
+	// overriding the ReadPartitions and WritePartitions
+	dc := s.testCluster.host.dcClient
+	dc.OverrideValue(s.T(), dynamicconfig.MatchingNumTaskqueueReadPartitions, partitions)
+	dc.OverrideValue(s.T(), dynamicconfig.MatchingNumTaskqueueWritePartitions, partitions)
+
+	tl := "backlog-counter-task-queue"
+	tq := &taskqueuepb.TaskQueue{Name: tl, Kind: enumspb.TASK_QUEUE_KIND_NORMAL}
+	identity := "worker-multiple-tasks"
+
+	for i := 0; i < workflows; i++ {
+
+		id := uuid.New()
+		wt := "functional-workflow-multiple-tasks"
+		workflowType := &commonpb.WorkflowType{Name: wt}
+
+		request := &workflowservice.StartWorkflowExecutionRequest{
+			RequestId:           uuid.New(),
+			Namespace:           s.namespace,
+			WorkflowId:          id,
+			WorkflowType:        workflowType,
+			TaskQueue:           tq,
+			Input:               nil,
+			WorkflowRunTimeout:  durationpb.New(20 * time.Second),
+			WorkflowTaskTimeout: durationpb.New(3 * time.Second),
+			Identity:            identity,
+		}
+
+		resp0, err0 := s.engine.StartWorkflowExecution(NewContext(), request)
+		s.NoError(err0)
+
+		we := &commonpb.WorkflowExecution{
+			WorkflowId: id,
+			RunId:      resp0.RunId,
+		}
+
+		s.EqualHistoryEvents(`
```

Comment:
not sure if this check is relevant or needed here.

Reply:
Removed.